### DISCUSSION TITLE
Installation and setup

### DISCUSSION THREAD

**Discussion by mmsarhan** [muscbridge/PyDKE#271] [Technical Support]
I am having trouble installing all the prerequisite programs before installing pydesigner. I was able to install acaconda and create a python environment with python 3.7. The issue I keep running into is that FSL will not download properly. It will get to the post installation set and fail. I am not sure if I am setting my program wrong or if it is something else. Hopefully someone can help. Thank you.

**Comment by TheJaeger** [comment]
Hey @mmsarhan, could you post the error from your FSL installation? It would also be helpful if you post your OS and machine information.

**Reply by mmsarhan** [reply]
Hello @TheJaeger 
Below is the message that I received. I am using a VM with Centos 7. 

Stage 1
By installing this python distribution you agree to the license terms in
/usr/local/fsl/fslpython/LICENSE.txt
100%
Stage 2
100%
Post install setup failed
[OK] Post installation setup complete
Setting up FSL software...
[OK] User profile updated with FSL settings, you will need to log out and back in to use the FSL tools.
To use FSLView please install the PNG12 and MNG libraries with: sudo yum install libpng12 libmng

**Reply by TheJaeger** [reply]
I believe it should be fine even if post-install scripts fail, you can repeat those steps manually. The most important tasks this post-install script does are configure FSL variables and PATH. You can do this manually on CentOS by adding the following lines to `/etc/profile.d`:

```
FSLDIR=/usr/local/fsl
FSLOUTPUTTYPE=NIFTI_GZ
PATH=$PATH:$FSLDIR/bin
LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$FSLDIR
```

Then logout and login to your VM. FSL commands should then work.

EDIT: Alternatively, you may pull a pre-built Docker container containing FSL, MRtrix3, and PyDesigner called [NeuroDock](https://hub.docker.com/r/dmri/neurodock/tags). You can also download the PyDesigner repo/release, `cd` into the `PyDesigner/docker`, and build the image manually with

```
docker build -t dmri/neurodock:v1.0-RC11 -f Dockerfile-release .
```

If you prefer Singularity over Docker, you may build the same file with Singularity instead.

**Comment by mmsarhan** [comment]
Where would I find /etc/profile.d:

Also are those command all one command or each a different command I have to run?

Mutaz M. Sarhan, MS, BS
Research Specialist II
Laboratory for Precision Brain Imaging
Department of Neuroscience
Medical University of South Carolina


From: Siddhartha Dhiman ***@***.***>
Date: Wednesday, December 1, 2021 at 2:54 PM
To: m-ama/PyDesigner ***@***.***>
Cc: Sarhan, Mutaz ***@***.***>, Mention ***@***.***>
Subject: Re: [m-ama/PyDesigner] Installation and setup (Discussion #271)

CAUTION: External

I believe it should be fine even if post-install scripts fail, you can repeat those steps manually. The most important tasks this post-install script does are configure FSL variables and PATH. You can do this manually on CentOS by adding the following lines to /etc/profile.d:

FSLDIR=/usr/local/fsl

FSLOUTPUTTYPE=NIFTI_GZ

PATH=$PATH:$FSLDIR/bin

LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$FSLDIR

Then logout and login to your VM. FSL commands should then work.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/m-ama/PyDesigner/discussions/271*discussioncomment-1731155__;Iw!!Ab1_Rw!RvYS8gt8uWwHhnaQkXarBH4P4SRWXxFKG2wo6mKsog2dNFbpRgEja0dyqCBGjg$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AWXURMQI3HZYSO57GM6XTR3UOZ4NTANCNFSM5JFBMWCQ__;!!Ab1_Rw!RvYS8gt8uWwHhnaQkXarBH4P4SRWXxFKG2wo6mKsog2dNFbpRgEja0eFozSAGg$>.
Triage notifications on the go with GitHub Mobile for iOS<https://urldefense.com/v3/__https:/apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675__;!!Ab1_Rw!RvYS8gt8uWwHhnaQkXarBH4P4SRWXxFKG2wo6mKsog2dNFbpRgEja0exo0XEQw$> or Android<https://urldefense.com/v3/__https:/play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign*3Dnotification-email*26utm_medium*3Demail*26utm_source*3Dgithub__;JSUlJSU!!Ab1_Rw!RvYS8gt8uWwHhnaQkXarBH4P4SRWXxFKG2wo6mKsog2dNFbpRgEja0dxLX3jBg$>.

**Reply by TheJaeger** [reply]
`/etc/profile.d` is exactly where you would find it, it's the path of the file you need to edit.

Those are not commands to run, instead you'd paste them into `/etc/profile.d`

**Reply by mmsarhan** [reply]
I just pasted them into the path. FSL now opens, but I still do not have access to fsleyes. I also cannot use any of the python commands such fslpython. When I run fslpython is says bash: fsl/python: command not found ...

**Reply by mmsarhan** [reply]
Hello @TheJaeger ,
I redid the post installation as the FSL Wiki stated that if the imcp is not shown in the file then something went wrong with the installation. This is the message i received when I redid everything. 
DKE) [blab@localhost ~]$ sudo $FSLDIR/etc/fslconf/fslpython_install.sh -f $FSLDIR
Stage 1
By installing this python distribution you agree to the license terms in
/usr/local/fsl/fslpython/LICENSE.txt
100%
Stage 2
100%
/usr/local/fsl/etc/fslconf/fslpython_install.sh: line 245:  5918 Killed                  FSLDIR=$fsl_dir "${miniconda_bin_dir}/conda" env create -f "${script_dir}/fslpython_environment.yml" 2>> "${miniconda_install_log}"
      5919 Done                    | ${script_dir}/progress.sh 400 ${quiet} >> "${miniconda_install_log}"

----- ERROR -----
Failed to create FSL Python environment - see /tmp/fslpythonkXUF/fslpython_miniconda_installer.log for details

(DKE) [blab@localhost ~]$

**Reply by TheJaeger** [reply]
Did you install FSL with a Conda environment? The problem may be that you ran the FSL installation script from within your Python 3.7 Conda environment, which is the reason why FSL's post-installation scripts targeting Python failed.

You'll need to run post-installation script again from system-level Python 2.7 (FSL doesn't work with Python 3) instead of a Conda environment.

**Reply by mmsarhan** [reply]
@TheJaeger 
I did the installation from a anaconda environment with python 3.7. Should I deactivate my environment and run fsl and then see if everything works? I am very confused on the exact procedure of how everything needs to be installed to run pydesigner. Can you give me a protocol or some directions on how to download everything? Thank you.

**Reply by TheJaeger** [reply]
You're correct, you need to deactivate the environment and install FSL with Python 2. The command you need to parse for this is

```
python2 /PATH_TO_fslinstaller.py 
```

Examples for the commands we use to install FSL, MRtrix3, and PyDesigner can be found in the [NeuroDock Dockerfile](https://github.com/m-ama/PyDesigner/blob/master/Dockerfile)

**Reply by mmsarhan** [reply]
@TheJaeger 
I installed FSL after I deactivated the condo environment and the same error message came up. FSLeye also did not work and I had to use the command "sudo yum install" to install fslview. Will everything work out if I run the rest of the programs? Also, does MRtrix and pydesigner have to be run within the conda environment or at the system?

**Reply by mmsarhan** [reply]
@TheJaeger I continue with the installation of MRtrix within the python environment and pydesigner. Pydesigner did not download correctly and I received this error message. Is there anywy to fix it and make it run properly.

-Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Icvxpy/cvxcore/src/ -Icvxpy/cvxcore/python/ -Icvxpy/cvxcore/include/ -I/home/blab/anaconda3/envs/DKE/include/python3.7m -I/tmp/pip-build-env-eult6pym/overlay/lib/python3.7/site-packages/numpy/core/include -c cvxpy/cvxcore/src/cvxcore.cpp -o build/temp.linux-x86_64-3.7/cvxpy/cvxcore/src/cvxcore.o -O3 -std=c++11 -Wall -pedantic -Wextra -Wno-unused-parameter
  gcc: error trying to exec 'cc1plus': execvp: No such file or directory
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for cvxpy
  Building wheel for scs (PEP 517) ... done
  Created wheel for scs: filename=scs-2.1.4-cp37-cp37m-linux_x86_64.whl size=335913 sha256=dda1c6776969e62d7fdb46cec85d0669639aaff8985b3946972e837a6298bd22
  Stored in directory: /home/blab/.cache/pip/wheels/d6/58/0b/a55ff6d4e2da4c9776e363a778ea13e4068037a7be82eb85e0
Successfully built PyDesigner scs
Failed to build cvxpy
ERROR: Could not build wheels for cvxpy which use PEP 517 and cannot be installed directly
(DKE) [blab@localhost PyDesigner-master]$

**Reply by TheJaeger** [reply]
Please see issue [#242](https://github.com/m-ama/PyDesigner/issues/242) for a fix.

**Reply by mmsarhan** [reply]
@TheJaeger 
pydesigner installed successful. How do I make sure everythin is running correctly?

**Reply by TheJaeger** [reply]
You can run the command `pydesigner --version`. It should output the version if FSL, MRtrix, and PyDesigner are installed correctly.

**Reply by mmsarhan** [reply]
I jut ran the command and it gave me this version so Im assuming everything should be downloaded correctly?
(DKE) [blab@localhost PyDesigner-master]$ pydesigner --version
v1.0-RC11

**Reply by TheJaeger** [reply]
There you go, it's ready for processing.
